source/_patterns/01-elements/input/input.scss

```diff
@@ -1,50 +1,145 @@
+	&[data-icon-before] {
+		--elm-input-padding-left: #{calc(2 * $db-spacing-fixed-sm + 0.425rem)};
+
+		input {
+			padding-left: var(--elm-input-padding-left);
+		}
+	}
+
+	&[data-icon-after] {
+		--elm-input-padding-right: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem
+			)};
+
+		input {
+			padding-right: var(--elm-input-padding-right);
+		}
+
+		.elm-state-icon::after {
+			margin-right: calc(0.75rem + $db-spacing-fixed-xs);
+		}
+	}
+
+	&[data-variant][data-icon-after] {
+		--elm-input-padding-right: #{calc(
+				2 * #{$db-spacing-fixed-sm} + 0.425rem + 0.75rem + #{$db-spacing-fixed-xs}
+			)};
+
+		input {
+			padding-right: var(--elm-input-padding-right);
+		}
+	}
+
+	&::before,
+	&::after {
+		@extend %db-body-sm;
+		--icon-font-family: var(--db-base-icon-font-family);
+		--icon-font-size: var(--db-base-icon-font-size);
+
+		height: var(--db-sizing-md);
+		display: flex;
+		align-items: center;
+		position: absolute;
+		top: 0;
+		padding: 0 $db-spacing-fixed-xs;
+	}
+
+	&::after {
+		right: 0;
+	}
+}
 
-	padding: $db-spacing-fixed-regular-xs $db-spacing-fixed-regular-m;
+.elm-input input {
+	@extend %db-bg-neutral-transparent-semi;
 
+	border: none;
+	border-radius: 4px;
+	height: $db-sizing-md;
+	max-width: 100%;
 	width: 100%;
+	padding: $db-spacing-fixed-xs var(--elm-input-padding-right)
+		$db-spacing-fixed-xs var(--elm-input-padding-left);
 
-	&:not(:placeholder-shown) {
-		padding-bottom: 0;
-		padding-top: to-rem($pxValue: 16);
+	& + label {
+		position: absolute;
+		display: flex;
+		align-items: center;
+		font-size: var(--db-type-body-font-size-md);
+		color: $db-colors-secondary-enabled;
+
+		transform-origin: left;
+		transform: translate(
+			var(--elm-input-padding-left),
+			calc(-1 * var(--db-sizing-md))
+		);
+		transition: transform 0.2s $db-transition-functional-timing;
+
+		height: $db-sizing-md;
+		pointer-events: none;
 	}
 
-	// * TODO: to get evaluated
-	// display: inline-flex;
-	// align-items: center;
-	// justify-content: flex-start;
-	// max-width: 100%;
+	& ~ .description {
+		@extend %db-body-xs;
+		margin: $db-spacing-fixed-2xs $db-spacing-fixed-xs 0;
+
+		display: flex;
+		align-items: flex-start;
+		justify-content: space-between;
+	}
 
-	// TODO: Evaluate whether those could get moved to ../_form-elements.scss
 	&:focus,
-	&:hover {
+	&:not(:placeholder-shown) {
 		&:not(:disabled) {
-			--formElement---borderColor: #{$db-colors-secondary-enabled};
+			padding-bottom: calc(
+				$db-spacing-fixed-xs - var(--db-input-focus-opacity) *
+					$db-spacing-fixed-xs
+			);
+			padding-top: $db-spacing-fixed-xs;
+
+			+ label {
+				@include label-focus-animation();
+			}
 		}
 	}
 
 	// TODO: Evaluate whether those could get moved to ../_form-elements.scss
 	&:disabled,
 	fieldset:disabled & {
 		&,
-		& + .elm-label {
+		& + label {
 			// Decided against cursor: not-allowed, compare to e.g. https://phabricator.wikimedia.org/T121960
-			opacity: 0.4;
+			opacity: 0.25;
+		}
+
+		& + .description {
```

Comment:
In between the input and the description there's most likely the label so we would need the tilde instead of plus here, wouldn't we ? And is this an actual decision to visually hide the text, but keep it's space ?

Reply:
changed in https://github.com/db-ui/core/pull/198/commits/9291c0e751599052a5c87dc858f92b0e719c1eaa